c10/util/TypeIndex.h

```diff
@@ -96,7 +96,12 @@ inline constexpr string_view extract(
 
 template <typename T>
 inline C10_TYPENAME_CONSTEXPR c10::string_view fully_qualified_type_name_impl() {
-#if defined(_MSC_VER) && !defined(__clang__)
+#if defined(__NVCC__)
```

Comment:
This should only apply to NVCC with MSVC, right?

Reply:
You might be right. Probably my changes work on my machine because I use only MSVC. I can add the __NVCC__ check nested inside the original _MSC_VER check

Reply:
Please do that.

Reply:
Done. But getting failed checks regarding linux xenial\bionic due to merging! what to do now?
![Capture](https://user-images.githubusercontent.com/75228100/103417570-2895a380-4b59-11eb-81c2-0b97dc241924.PNG)